js/src/layers/ImageServiceLayer.js

```diff
@@ -0,0 +1,113 @@
+      compressionQuality: '',
+      bandIds: null,
+      time: null,
+      renderingRule: null,
+      mosaicRule: null,
+      transparent: false,
+      crs: null,
+      _url: '',
+      _bounds: null,
+    };
+  }
+}
+
+export class LeafletImageServiceLayerView extends imageoverlay.LeafletImageOverlayView {
+  create_obj() {
+    this.model._url = this.model.get('url') + '/exportImage' + this.buildParams()
+    this.model._bounds = this._map.getBounds()
+    this.obj = L.ImageOverlay(this.model._url, this.model._bounds)
+  }
+
+  model_events() {
+    super.model_events();
+    for (var option in this.get_options()) {
+      this._map.on('change:' + option, () => {
+        this.model._url = this.model.get('url') + '/exportImage' + this.buildParams()
+        this.model._bounds = this._map.getBounds()
+        this.refresh();
+      });
+    }
+  }
+
+  model_epsg() {
+    // get the EPSG code of the current map
+    var crs = proj.getProjection(this.model.get('crs'))
+    var spatial_reference = parseInt(crs.code.split(':')[1], 10);
+    return spatial_reference
+  }
+
+  buildParams () {
+    // required parmeters for query
+    var params = {
+      f: this.model.get('f'),
+      format: this.model.get('format'),
+      bbox: this._map.getBounds(),
+      size: this._map.getSize(),
+      bboxSR: 4326,
+      imageSR: this.model_epsg()
+    };
+    // append optional parameters
+    // image transparency
+    if (this.model.get('transparent')) {
+      params.transparent = this.model.get('transparent');
+    }
+    // data type of the raster image
+    if (this.model.get('pixelType')) {
+      params.pixelType = this.model.get('pixelType');
+    }
+    // pixel value or list of pixel values representing no data
+    if (this.model.get('noData')) {
+      params.noData = this.model.get('noData').join(',');
+    }
+    // how to interpret no data values
+    if (this.model.get('noDataInterpretation')) {
+      params.noDataInterpretation = this.model.get('noDataInterpretation');
+    }
+    // resampling process for interpolating the pixel values
+    if (this.model.get('interpolation')) {
+      params.interpolation = this.model.get('interpolation');
+    }
+    // lossy quality for image compression
+    if (this.model.get('compressionQuality')) {
+      params.compressionQuality = this.model.get('compressionQuality');
+    }
+    // order of bands to export for multiple band images
+    if (this.model.get('bandIds')) {
+      params.bandIds = this.model.get('bandIds').join(',');
+    }
+    // time instance or extent for image
+    if (this.model.get('time')) {
+      params.time = this.model.get('time').join(',');
+    }
+    // rules for rendering
+    if (this.model.get('renderingRule')) {
+      params.renderingRule = JSON.stringify(this.model.get('renderingRule'));
+    }
+    // rules for mosaicking
+    if (this.model.get('mosaicRule')) {
+      params.mosaicRule = JSON.stringify(this.model.get('mosaicRule'));
+    }
```

Comment:
We have a method for getting all options at once (properties tagged with `.tag(sync=True, o=True)` python side): `this.get_options()`, it will automatically translate properties names from snake_case to camelCase.

So I could suggest doing something like:

```javascript
var params = {
    bbox: this._map.getBounds(),
    size: this._map.getSize(),
    bboxSR: 4326,
    imageSR: this.model_epsg(),
    ...this.get_options()
};
if (params['bandIds']) {
    params['bandIds'] = params['bandIds'].join(',');
}
// etc
```

Reply:
thanks @martinRenou!  those changes are in :+1: